lib/stores/S3Store.js

```diff
@@ -276,57 +276,81 @@ class S3Store extends DataStore {
                 log(`[${metadata.file.id}] finished uploading part #${current_part_number}`);
 
                 return data.ETag;
-            })
-            .catch((err) => {
-                throw err;
             });
     }
 
     /**
-     * Processes the following steps:
-     * - uploads a part (stream) to s3
-     * - finishes the upload in case offset === length
-     *
-     * @param {Object}                metadata            upload metadata
-     * @param {http<IncomingMessage>} req                 incoming request
-     * @param {Number}                current_part_number number of the current part/chunk
-     * @return {Promise<Number>}                         which resolves with the current offset
-     * @memberof S3Store
-     */
-    _processUpload(metadata, req, current_part_number) {
-        return this._uploadPart(metadata, req, current_part_number)
-            .then(() => this.getOffset(metadata.file.id, true))
-            .then((current_offset) => {
-                if (parseInt(metadata.file.upload_length, 10) === current_offset.size) {
-                    return this._finishMultipartUpload(metadata, current_offset.parts)
-                        .then((location) => {
-                            log(`[${metadata.file.id}] finished uploading: ${location}`);
-
-                            this.emit(EVENTS.EVENT_UPLOAD_COMPLETE, {
-                                file: Object.assign({}, metadata.file, { location }),
-                            });
-
-                            this._clearCache(metadata.file.id);
-
-                            return current_offset.size;
-                        })
-                        .catch((err) => {
-                            throw err;
+   * uploads a stream to s3 using multiple parts
+   *
+   * @param {Object}                metadata            upload metadata
+   * @param {http<IncomingMessage>} req                 incoming request
+   * @param {Number}                currentPartNumber number of the current part/chunk
+   * @return {Promise<Number>}                         which resolves with the current offset
+   * @memberof S3Store
+   */
+    _processUpload(metadata, req, currentPartNumber) {
+        return new Promise((resolve, reject) => {
+            const splitter = new FileStreamSplitter(
+                {},
+                { maxChunkSize: this.part_size, directory: '.' }
```

Comment:
> directory: '.'

Does this mean that the temporary chunk files will be written in the current working directory? Would it be better to save them directly in the system's default temporary directory?